src/test/java/com/hotels/beeju/HiveServer2JUnitRuleTest.java

```diff
@@ -0,0 +1,265 @@
+ * limitations under the License.
+ */
+package com.hotels.beeju;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+import org.apache.hadoop.hive.metastore.api.Database;
+import org.apache.hadoop.hive.metastore.api.FieldSchema;
+import org.apache.hadoop.hive.metastore.api.NoSuchObjectException;
+import org.apache.hadoop.hive.metastore.api.Partition;
+import org.apache.hadoop.hive.metastore.api.SerDeInfo;
+import org.apache.hadoop.hive.metastore.api.StorageDescriptor;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class HiveServer2JUnitRuleTest {
+
+  private static final String DATABASE = "my_test_db";
+
+  public @Rule HiveServer2JUnitRule server = new HiveServer2JUnitRule(DATABASE);
+
+  @Before
+  public void init() throws Exception {
+    Class.forName(server.driverClassName());
+  }
+
+  @Test
+  public void databaseNane() {
+    assertThat(server.databaseName(), is(DATABASE));
+  }
+
+  @Test
+  public void showCreateTable() throws Exception {
+    String tableName = "my_table";
+    Table table = createUnpartitionedTable(DATABASE, tableName);
+
+    StringBuilder showCreateTable = new StringBuilder(256);
+    try (Connection connection = DriverManager.getConnection(server.connectionURL());
+        Statement statement = connection.createStatement()) {
+      String showHql = String.format("SHOW CREATE TABLE %s.%s", DATABASE, tableName);
+      ResultSet result = statement.executeQuery(showHql);
+      while (result.next()) {
+        showCreateTable.append(result.getString(1)).append("\n");
+      }
+      result.close();
+    }
+    String expectedShowCreateTable = new StringBuilder(256)
+        .append("CREATE TABLE `my_test_db." + tableName + "`(\n")
+        .append("  `id` int, \n")
+        .append("  `name` string)\n")
+        .append("ROW FORMAT SERDE \n")
+        .append("  'org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe' \n")
+        .append("STORED AS INPUTFORMAT \n")
+        .append("  'org.apache.hadoop.mapred.TextInputFormat' \n")
+        .append("OUTPUTFORMAT \n")
+        .append("  'org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat'\n")
+        .append("LOCATION\n")
+        .append("  'file:" + server.temporaryFolder.getRoot() + "/" + DATABASE + "/my_table'\n")
+        .append("TBLPROPERTIES (\n")
+        .append("  'transient_lastDdlTime'='" + table.getParameters().get("transient_lastDdlTime") + "')\n")
+        .toString();
+    assertThat(showCreateTable.toString(), is(expectedShowCreateTable));
```

Comment:
The point of this test is to test the output of the show command - the table is created using the Thrift API. The only way this could change is if the grammar changes. Let's revisit later if necessary, otherwise I'll remove the test